pkg/ui/src/views/statements/planView.tsx

```diff
@@ -16,115 +16,314 @@ import _ from "lodash";
-          </span>
-        ))}
-      </span>
-      { renderChildren(node.children) }
-    </div>
-  );
+interface PlanNodeHeaderProps {
+  title: string;
+  subtitle: string;
+  warn: boolean;
 }
 
-function renderChildren(children: cockroach.sql.IExplainTreePlanNode[]|null): React.ReactElement<PlanNodeProps> {
-  if (!children) {
-    return null;
+// planNodeHeaderProps looks at a node's name and attributes and determines if:
+// 1) We should warn the user about this particular node,
+// 2) We should include any attribute details in the collapsed node view.
+export function planNodeHeaderProps(node: FlatPlanNode): PlanNodeHeaderProps {
+  let title = node.name;
+  let subtitle = null;
+  let warn = false;
+
+  switch (node.name) {
+    case "scan":
+      if (_.find(node.attrs, {key: "spans", value: "ALL"})) {
+        const tableAttr = _.filter(node.attrs, (attr) => ( attr.key === "table"));
+        if (tableAttr.length >= 1) {
+          title = "table scan";
+          warn = true;
+          subtitle = tableAttr[0].value;
+          const ampersandPosition = subtitle.indexOf("@");
+          if (ampersandPosition > 0) {
+            subtitle = subtitle.substring(0, ampersandPosition);
+          }
+        }
+      }
+      break;
+    case "join":
+      const typeAttr = _.filter(node.attrs, (attr) => ( attr.key === "type"));
+      title = typeAttr[0].value + " " + node.name;
+      break;
+    default:
+      break;
   }
-  return (
-    <ul>
-      {children.map((child, idx) => (
-        <li>
-          <PlanNode key={idx} node={child} />
-        </li>
-      ))}
-    </ul>
-  );
+
+  return {
+    title, subtitle, warn,
+  };
 }
 
-function renderAttrValueList(values: string[]) {
-  return (
-    <span className="plan-node__attr-value-list">
-      [
-      {intersperse(
-        values.map((value, idx) => (
-          <span className="plan-node__attr-value" key={idx}>{value}</span>
-        )),
-        <span>, </span>,
-      )}
-      ]
-    </span>
-  );
+/* ************************* PLAN NODES ************************* */
+
+interface PlanNodeDetailProps {
+  node: FlatPlanNode;
 }
 
-interface CollapsedPlanAttr {
-  key: string;
-  value: string[];
+interface PlanNodeDetailState {
+  expanded: boolean;
 }
 
-export function collapseRepeatedAttrs(attrs: ExplainTreePlanNode_IAttr[]): CollapsedPlanAttr[] {
-  if (!attrs) {
-    return [];
+class PlanNodeDetails extends React.Component<PlanNodeDetailProps, PlanNodeDetailState> {
+  constructor(props: PlanNodeDetailProps) {
+    super(props);
+    this.state = {
+      expanded: false,
+    };
+  }
+
+  toggleExpanded = () => {
+    this.setState(state => ({
+      expanded: !state.expanded,
+    }));
   }
-  const collapsed: { [key: string]: CollapsedPlanAttr } = {};
 
-  attrs.forEach((attr) => {
-    if (attr.key === null || attr.value === null) {
-      return;
+  renderPlanNodeHeader(props: PlanNodeHeaderProps) {
+    return <span>
```

Comment:
syntax nit: also use parens here, and indent all these children at the same level